renku/core/dataset/dataset.py

```diff
@@ -472,17 +471,16 @@ def remove_files(dataset):
     try:
         importer = provider.get_importer(uri, gitlab_token=gitlab_token, **kwargs)
         provider_dataset: ProviderDataset = importer.fetch_provider_dataset()
-        is_latest_version = importer.is_latest_version()
+
+        if not importer.provider_dataset_files:
+            raise errors.ParameterError(f"Dataset '{uri}' has no files.")
+        if not importer.is_latest_version():
+            communication.warn(f"Newer version found at {importer.latest_uri}")
```

Comment:
I just remembered I always found this message a bit odd. It just informs the user that there is a new version that could be downloaded instead of the one they're currently downloading. But the way it's worded, it always left me confused, especially because the confirm_download method will print `Do you wish to download this version?`

So
```
Newer version found at .....
Do you wish to download this version?
```
implies "Do you want to download the new version instead of the one you're trying to download?" even though that's not what it does, it's purely a hint that there is a newer version. We should change that

Reply:
I moved the new version warning after the download confirmation. This makes it clear that which version will be downloaded.